### PR TITLE
added csrf protection

### LINKED ISSUE #2331: Add CSRF to /api/delete-shot
I'm pretty sure `/api/delete-shot` has a danger of CSRF.  We should sign those requests in some fashion.

### PR DESCRIPTION
**Pull request by dannycoates**
fixes #2331 and other important routes

### CODE REVIEW

File: server/src/pages/shot/controller.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ exports.changeShotExpiration = function (shot, expiration) {
       location.reload();
     };
   }
-  req.send(`id=${encodeURIComponent(shot.id)}&expiration=${encodeURIComponent(expiration)}`);
+  req.send(`id=${encodeURIComponent(shot.id)}&expiration=${encodeURIComponent(expiration)}&_csrf=${model.csrfToken}`);
```
Comment (ianb) [inline server/src/pages/shot/controller.js]: I like to use `encodeURIComponent()` everywhere even when I'm pretty sure it won't be needed.

File: server/src/reactrender.js [hunk 2]
```diff
@@ -9,18 +9,21 @@ exports.render = function (req, res, page) {
     model.backend = req.backend;
     let jsonModel = model.jsonModel || model;
     let serverModel = model.serverModel || model;
+    let csrfToken = req.csrfToken && req.csrfToken();
```
Comment (ianb) [inline server/src/reactrender.js]: When would this be not be there?

File: server/src/server.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ const {
 const dbschema = require("./dbschema");
 const express = require("express");
 const bodyParser = require('body-parser');
+const csrf = require("csurf");
```
Comment (ianb) [inline server/src/server.js]: Renaming it on import seems odd...?

File: package.json [hunk 4]
```diff
@@ -13,6 +13,7 @@
     "convict": "3.0.0",
     "cookies": "0.7.0",
     "core-js": "2.4.1",
+    "csurf": "^1.9.0",
```
Comment (ianb) [inline package.json]: Should be exact version

File: server/src/reactrender.js (hunk 2 above)
Comment (dannycoates) [inline server/src/reactrender.js]: it won't be there on routes that don't use the `csrfProtection` middleware

File: server/src/server.js (hunk 3 above)
Comment (dannycoates) [inline server/src/server.js]: I guess I did this cuz the csurf readme does

**Pull request merged**